### ISSUE TITLE
Chromium won't start

### ISSUE THREAD

**Issue by wooque** [linuxmint/lmde-3-cinnamon-beta#3]
Error message in terminal
```
vuk@vuk-pc:~$ chromium 
(chromium:3130): GLib-GIO-ERROR **: Settings schema 'org.cinnamon.muffin' does not contain a key named 'button-layout'
Trace/breakpoint trap
```
Installed in Virtualbox virtual machine

**Comment by fblais** [comment]
Same bug here.

**Comment by clefebvre** [comment]
oh.. this is weird :) Thanks, we'll check that out.

**Comment by jspoonamore** [comment]
I've got LMDE3 installed in VB as well.  So far, all software installed has ran just fine except for Chromium.  Same issue user "wooque" is having.

**Comment by SHrenov** [comment]
+1

Fresh LMDE3 Beta x64 installation, package chromium from software manager, When i launch chromium i get:

`(chromium:2609): GLib-GIO-ERROR **: Settings schema 'org.cinnamon.muffin' does not contain a key named 'button-layout'
Ловушка трассировки/останова`

Also, in package manager chromium on it's page come with picture from chromium-bsu page (package with arcade game). Not a big bug, but very funny.

**Comment by xenopeek** [comment]
**In short:** this is because the Debian Stretch (old) Muffin version has a **button-layout** key in its gsettings schema **org.cinnamon.muffin** that the LMDE 3 (current) Muffin version doesn't have _and_ Chromium on Debian Stretch is hardcoded to use that older Muffin schema.

I compared output of `gsettings list-recursively | grep button-layout` on LMDE 3, LMDE 2 and Linux Mint 19.

On LMDE 2 it shows:
```
org.cinnamon.muffin button-layout ':minimize,maximize,close'
org.gnome.desktop.wm.preferences button-layout 'menu:minimize,maximize,close'
org.cinnamon.desktop.wm.preferences button-layout 'menu:minimize,maximize,close'
```
On LMDE 3 and Linux Mint 19 it shows the same **except the first line** is missing. Chromium on Linux Mint 19 doesn't have an issue with this but Chromium on LMDE 3 indeed does.

Checking the Chromium sources on Debian Stretch it shows this is hardcoded specifically for Cinnamon in the file `chrome/browser/ui/libgtkui/settings_provider_gsettings.cc`:
```c++
const char kCinnamonPreferencesSchema[] = "org.cinnamon.muffin";
[...]
const char kButtonLayoutKey[] = "button-layout";
```

The key can't be added manually with gsettings as there exists no schema that has it on LMDE 3. I doubt Chromium hardcoding will be fixed in Debian Stretch.

A possible solution for LMDE 3 would be to add the schema back to the Muffin package for LMDE 3 and patch the Cinnamon Settings > Windows code so that the button layout in /org/cinnamon/desktop/wm/preferences/button-layout is always also set on  /org/cinnamon/muffin/button-layout?

**Comment by fblais** [comment]
Thanks.
I’ll probably try to install Chrome instead of Chromium, then...  :(

**Comment by fblais** [comment]
Update: Chrome is working correctly.

**Comment by clefebvre** [comment]
Debian won't fix it because it doesn't affect them. We've established that with NM already.

**Comment by clefebvre** [comment]
The simplest solution might be to add that key somewhere in debian-system-adjustments.

**Comment by clefebvre** [comment]
sorry, closed by mistake :)

**Comment by clefebvre** [comment]
Fixed in muffin 3.8.3.
